src/otk/parserstate.py

```diff
@@ -0,0 +1,17 @@
+class ParserState:
+
+    def __init__(self, path, defines):
+        self.path = path
+        self.defines = defines
+
+    def new(self, *, path=None, defines=None) -> "ParserState":
```

Comment:
Since this is all pretty close to `contextvar`'s, perhaps we can call this `def copy(...)`?

Reply:
Renamed to `State.new()`